app/controllers/site_controller.rb

```diff
@@ -734,6 +748,8 @@ def download_file
     elsif !@study.can_download?(current_user)
       redirect_to merge_default_redirect_params(view_study_path(accession: @study.accession, study_name: @study.url_safe_name), scpbr: params[:scpbr]),
                   alert: 'You do not have permission to perform that action.' and return
+    elsif @study.has_download_agreement? && !@study.download_agreement.user_accepted?(current_user)
```

Comment:
This multi-part check is done several times in various files.  Wrapping it in a simple helper function would help maintainability.

Reply:
Agreed, but the issue is that since we are attempting to intercept and redirect in that method, if this is pulled out into a helper method, the `and return` no longer functions in the manner that we need (execution continues in the main method, which is the exact opposite of what we want).  I had tried this previously and wasn't able to figure out hte correct way to do it.  While there likely is a solution to the problem, it didn't seem feasible to spend additional time to figure it out given the nature/timeline of this feature.

Reply:
Agree that the pulling out the `return` would change scopes in an undesirable way.  But I'm imagining a simple helper method in study.rb `download_agreement_needs_acceptance?(user)` that just returns `has_download_agreement? && !download_agreement.user_accepted?(current_user)`. Would that work?

Reply:
I did a little reading this morning and found a possible solution that I'm going to try - we can extract this logic into a helper method and then call `; return if performed?`.  This leverages `ActionController::Metal#performed?` which checks if `render` or `redirect_to` has been called.

Reply:
6449efb